tests/tests_geomstats/test_multinomial.py

```diff
@@ -1,153 +1,131 @@
-"""Unit tests for the categorical manifold."""
-
-import warnings
+"""Unit tests for the multinomial manifold."""
 
 import geomstats.backend as gs
-import tests.conftest
-from geomstats.information_geometry.multinomial import (
-    MultinomialDistributions,
-    MultinomialMetric,
-)
-
-
-class TestMultinomialDistributions(tests.conftest.TestCase):
-    """Class defining the categorical distributions tests."""
-
-    def setup_method(self):
-        """Define the parameters of the tests."""
-        gs.random.seed(0)
-        warnings.simplefilter("ignore", category=UserWarning)
-        self.dim = 3
-        self.n_draws = 2
-        self.multinomial = MultinomialDistributions(dim=self.dim, n_draws=self.n_draws)
-        self.metric = MultinomialMetric(self.dim, n_draws=self.n_draws)
-        self.n_points = 10
-        self.n_samples = 20
-
-    def test_random_point(self):
-        """Test random_point.
-
-        Test that the result belongs to the simplex.
-        """
-        point = self.multinomial.random_point(self.n_points)
-        result = gs.sum(point, -1)
-        expected = gs.ones(self.n_points)
-        self.assertAllClose(expected, result)
+from tests.conftest import Parametrizer, np_backend, pytorch_backend, tf_backend
+from tests.data.multinomial_data import MultinomialMetricTestData, MultinomialTestData
+from tests.geometry_test_cases import LevelSetTestCase, RiemannianMetricTestCase
 
-    def test_projection(self):
-        """Test projection.
+TF_OR_PYTORCH_BACKEND = tf_backend() or pytorch_backend()
+
+NOT_AUTOGRAD = tf_backend() or pytorch_backend() or np_backend()
+
+
+class TestMultinomialDistributions(LevelSetTestCase, metaclass=Parametrizer):
+    """Class defining the multinomial distributions tests."""
+
+    skip_test_extrinsic_after_intrinsic = True
+    skip_test_intrinsic_after_extrinsic = True
+    testing_data = MultinomialTestData()
+
+    def test_sample_shape(self, dim, n_draws, point, n_samples, expected):
+        self.assertAllClose(
+            self.Space(dim, n_draws).sample(point, n_samples).shape, expected
+        )
 
+    def test_projection(self, dim, n_draws):
```

Comment:
Probably we can abstract this test later (and also `test_to_tangent`). That is being passed as `smoke`, but in fact it is a `random` test.

Reply:
Yes, this comes from the fact that the tests were written in the old style format. I've spent some time to adapt them a little, but more needs to be done.